lib/LocalSigner.ts

```diff
@@ -12,30 +17,47 @@ export default class LocalSigner implements ISigner {
   /**
    * Creates a new local signer using the given private key.
    */
-  public static create (privateKey: JwkEs256k): ISigner {
+  public static create (privateKey: JwkEs256k | JwkEd25519): ISigner {
     return new LocalSigner(privateKey);
   }
 
-  private constructor (private privateKey: JwkEs256k) {
-    InputValidator.validateEs256kOperationKey(privateKey, OperationKeyType.Private);
+  private constructor (private privateKey: JwkEs256k | JwkEd25519) {
+    InputValidator.validateOperationKey(privateKey, OperationKeyType.Private);
   }
 
   public async sign (header: object, content: object): Promise<string> {
     const publicKeyJwk = {
       ...this.privateKey,
       d: undefined
     };
-    const key = await Secp256k1KeyPair.from({
-      type: 'JsonWebKey2020',
-      publicKeyJwk,
-      privateKeyJwk: this.privateKey
-    } as any);
-    const signer = key.signer();
-    const jwsSigner = await JWS.createSigner(signer, 'ES256K', {
-      detached: false,
-      header
-    });
-    const compactJws = await jwsSigner.sign({ data: content });
-    return compactJws;
+    if (IonKey.isJwkEs256k(publicKeyJwk)) {
+      const key = await Secp256k1KeyPair.from({
+        type: 'JsonWebKey2020',
+        publicKeyJwk,
+        privateKeyJwk: this.privateKey
+      } as any);
+      const signer = key.signer();
+      const jwsSigner = await JWS.createSigner(signer, 'ES256K', {
+        detached: false,
+        header
+      });
+      const compactJws = await jwsSigner.sign({ data: content });
+      return compactJws;
+    } else if (IonKey.isJwkEd25519(publicKeyJwk)) {
+      const key = await Ed25519KeyPair.from({
+        type: 'JsonWebKey2020',
+        publicKeyJwk,
+        privateKeyJwk: this.privateKey
+      } as any);
+      const signer = key.signer();
+      const jwsSigner = await JWS.createSigner(signer, 'EdDSA', {
```

Comment:
it might be worth a sanity test with https://github.com/panva/jose in this library to cover JWS interop, we have one here, but you can never test this kind of thing enough...  https://github.com/transmute-industries/verifiable-data/blob/main/packages/jose-ld/src/JWS/jose.test.ts